src/clr-addons/daterangepicker/validators/daterange-min-max.validator.ts

```diff
@@ -0,0 +1,99 @@
+import { Directive, forwardRef, Input, Optional } from '@angular/core';
+import { AbstractControl, NG_VALIDATORS, ValidationErrors, Validator } from '@angular/forms';
+
+import { NullableDaterange } from '../interfaces/daterange.interface';
+import { NullableDayModel } from '../models/day.model';
+import { DaterangeService } from '../providers/daterange.service';
+
+/**
+ * Validator for daterangepicker to validate that 'from' and 'to' dates are within the 'min' and 'max' values.
+ * Otherwise will throw corresponding `min` or `max` validation error.
+ *
+ * Validator is applied to the ClrDaterangepickerDirective with `min` or `max` attribute.
+ * Or when set manually with `clrDaterangeMinMax` attribute.
+ *
+ * Validator is active by default when applied. It can be disabled with: `[clrDaterangeMinMax]="false"`
+ */
+@Directive({
+  selector: '[clrDaterangepicker][min], [clrDaterangepicker][max], [clrDaterangeMinMax]',
+  providers: [
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => ClrDaterangeMinMaxValidator),
+      multi: true,
+    },
+  ],
+})
+export class ClrDaterangeMinMaxValidator implements Validator {
```

Comment:
Thanks for pointing me in the right direction.

Reply:
After thinking about this a bit, I disagree that there should be reactive form validations for these validators.

The order validator should always be applied. The active option is even questionable here. Why would someone disable this!

The min and max validator is only applied once the corresponding `min` or `max` attribute is set. 
The values from the attributes are used in the datepickers. Adding this as a reactive form validation would not update the datepickers. 
These validators also require access to the `DaterangeService` provider from the  `clrDaterangepicker` directive, which is not possible in reactive form validators.
Plus, this is in line with Clarity; there is only an `ClrDate` directive validator (no reactive form validator), that handles the `min` and `max` validation. This is done automatically once the `min` and/or `max` attribute is set. See https://github.com/vmware-clarity/ng-clarity/blob/0a3913cd9e9e0decbfd0c21af3acd2e0740a96c7/projects/angular/src/forms/datepicker/date-input.validator.ts
Angular build-in min and max validators are not related here, because the are based on numeric values, not dates.

The required validator is applied once the `required` attribute is set. 
This one might be the only candidate for a reactive form validator. Angular build-in required validator (`Validators.required`) already works, but doesn't check for empty 'from' and 'to' properties like this required validator. So that means if you want to use the full required validation in reactive form, you need both validators; `Validators.required` and our custom required validator (e.g. `DaterangeValidators.required`). This looks counterintuitive. I rather explain on the website they should use the `required` attribute.

Open for counter arguments.

Reply:
I understand and am fine with the proposed way to go for the required validation.